stages/test/test_cloud-init.py

```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/python3
+
+import os.path
+
+import pytest
+
+STAGE_NAME = "org.osbuild.cloud-init"
+
+
+# Test that the 'datasource_list' option is dumped as a single line
+# in the configuration file.
+@pytest.mark.parametrize("stage_module", [STAGE_NAME], indirect=True)
+def test_cloud_init_datasource_list_single_line(tmp_path, stage_module):
+    treedir = tmp_path / "tree"
+    confpath = treedir / "etc/cloud/cloud.cfg.d/datasource_list.cfg"
+    confpath.parent.mkdir(parents=True, exist_ok=True)
+
+    options = {
+        "filename": confpath.name,
+        "config": {
+            "datasource_list": [
+                "Azure",
+            ]
+        }
+    }
+
+    stage_module.main(treedir, options)
+    assert os.path.exists(confpath)
+
+    conf = None
```

Comment:
(nitpick) these lines could be sligtly more compact with the magic of pathlib.Path (that we now use)
```python
    assert confpath.exists()
    assert confpath.read_bytes() == b"datasource_list: [Azure]\n"
```

Reply:
sure, why not... :slightly_smiling_face:

Reply:
done